schema/manifest_test.go

```diff
@@ -113,9 +118,75 @@ func TestManifest(t *testing.T) {
 `,
 			fail: false,
 		},
+
+		// valid manifest accepts customized config media types
+		{
+			strict: false,
+			manifest: `
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "config": {
+    "mediaType": "application/vnd.customized.config+json",
+    "size": 1470,
+    "digest": "sha256:c86f7763873b6c0aae22d963bab59b4f5debbed6685761b5951584f6efb0633b"
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.customized.layer.tar+gzip",
+      "size": 148,
+      "digest": "sha256:c57089565e894899735d458f0fd4bb17a0f1e0df8d72da392b85c9b35ee777cd"
+    }
+  ]
+}
+`,
+			fail: false,
+		},
+
+		// expected failure strictly: unknown config.mediaType
+		{
+			strict: true,
+			manifest: `
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "config": {
+    "mediaType": "application/vnd.customized.config+json",
+    "size": 1470,
+    "digest": "sha256:c86f7763873b6c0aae22d963bab59b4f5debbed6685761b5951584f6efb0633b"
+  },
+  "layers": []
```

Comment:
On Thu, Nov 17, 2016 at 06:00:35PM -0800, xiekeyang wrote:
> … but current schema really allows empty layer slice (I've debug it). I think spec also allow empty…

I don't think the spec allows empty layers, based [on][1]:

> The array MUST have the base image at index 0.

#407, and #318 before it, are attempting to lift that restriction.  But I agree that you don't want to get involved in that discussion in this PR.  By using the usual single-entry `layers` array here, you would avoid getting involved.

[1]: https://github.com/opencontainers/image-spec/blame/de440dbfe41f35eb03117204bc99eeb0e6a186b8/manifest.md#L45